compatibility/compatibility-tests/src/intTest/java/org/projectnessie/tools/compatibility/tests/AbstractCompatibilityTests.java

```diff
@@ -115,9 +115,73 @@ void getDefaultBranchV2() throws Exception {
   }
 
   @Test
-  void getConfig() {
+  @VersionCondition(maxVersion = "0.45.0")
+  void getConfigV1WithoutV2Support() {
     NessieConfiguration config = api.getConfig();
-    assertThat(config).extracting(NessieConfiguration::getDefaultBranch).isEqualTo("main");
+    assertThat(config)
+        .extracting(
+            NessieConfiguration::getDefaultBranch,
+            NessieConfiguration::getMinSupportedApiVersion,
+            NessieConfiguration::getMaxSupportedApiVersion,
+            NessieConfiguration::getActualApiVersion,
+            NessieConfiguration::getSpecVersion)
+        .containsExactly("main", 1, 1, 0, null);
+  }
+
+  @Test
+  @VersionCondition(minVersion = "0.47.0")
+  void getConfigV1WithV2Support() {
+    NessieConfiguration config = api.getConfig();
+    assertThat(config)
+        .extracting(
+            NessieConfiguration::getDefaultBranch,
+            NessieConfiguration::getMinSupportedApiVersion,
+            NessieConfiguration::getMaxSupportedApiVersion,
+            NessieConfiguration::getActualApiVersion,
+            NessieConfiguration::getSpecVersion)
+        .containsExactly("main", 1, 2, 0, null);
+  }
+
+  @Test
+  @VersionCondition(minVersion = "0.47.0", maxVersion = "0.51.1")
+  void getConfigV2WithoutSpecWithoutActualVersion() {
+    NessieConfiguration config = apiV2.getConfig();
+    assertThat(config)
+        .extracting(
+            NessieConfiguration::getDefaultBranch,
+            NessieConfiguration::getMinSupportedApiVersion,
+            NessieConfiguration::getMaxSupportedApiVersion,
+            NessieConfiguration::getActualApiVersion,
+            NessieConfiguration::getSpecVersion)
+        .containsExactly("main", 1, 2, 0, null);
+  }
+
+  @Test
+  @VersionCondition(minVersion = "0.52.3", maxVersion = "0.57.0")
+  void getConfigV2WithSpecWithoutActualVersion() {
+    NessieConfiguration config = apiV2.getConfig();
+    assertThat(config)
+        .extracting(
+            NessieConfiguration::getDefaultBranch,
+            NessieConfiguration::getMinSupportedApiVersion,
+            NessieConfiguration::getMaxSupportedApiVersion,
+            NessieConfiguration::getActualApiVersion,
+            NessieConfiguration::getSpecVersion)
+        .containsExactly("main", 1, 2, 0, "2.0-beta.1");
+  }
+
+  @Test
+  @VersionCondition(minVersion = "0.59.0")
+  void getConfigV2WithSpecWithActualVersion() {
```

Comment:
This test currently doesn't run, but will when 0.59.0 is released and included in the tested versions.